core/src/wallet/ripple/api_impl/RippleLikeTransactionApi.cpp

```diff
@@ -234,6 +266,40 @@ namespace ledger {
             auto receiverHash = _receiver->getHash160();
             writer.writeVarInt(receiverHash.size());
             writer.writeByteArray(receiverHash);
+
+            if (!_memos.empty()) {
+                writer.writeByte(0xF9); // STI_ARRAY (Memos); Type Code = 15, Memos; Field ID = 9
+
+                for (auto& memo : _memos) {
+                    writer.writeByte(0xEA); // STI_OBJECT (Memo); Type Code = 10
+
+                    if (!memo.ty.empty()) {
+                        writer.writeByte(0x7C); // STI_VL (MemoType), 12
+                        auto written = writeLengthPrefix(writer, memo.ty.size());
+                        assert(written);
```

Comment:
Is this assert from cassert http://www.cplusplus.com/reference/cassert/assert/  ? If yes, what would happened in release mode if the error occured?

Reply:
It is.

In release mode, all `assert` calls disappear.

Reply:
So will have some unknow behavior in this case. Instead we can throw nice exception.

Reply:
If I were to redo the whole libcore parsing, it wouldn’t have asserts nor exceptions. But proper error handling. In this case, I think it’s okay (the rest of the serializer, for all other currencies, don’t even look at the values read back).

I don’t want to spend a week to redo all the parsers / serializers just because we know they could be enhanced. I think it’s time for another time. :)

Reply:
ok